lib/Windowing/WindowingProvider.js

```diff
@@ -0,0 +1,50 @@
+import React, { useState, useCallback, createContext } from 'react';
+import { node, number } from 'prop-types';
+import debounce from 'debounce';
+
+export const WindowingContext = createContext({});
+
+const WindowingProvider = ({
+  children,
+  containerHeight,
+  debounceTimer,
+  ...rest
+}) => {
+  const [scrollTop, setScrollTop] = useState(null);
+  const [height, setHeight] = useState(null);
+
+  const ref = useCallback(elem => {
+    if (elem !== null) {
+      setHeight(elem.getBoundingClientRect().height || containerHeight);
+      setScrollTop(elem.scrollTop);
+
+      const debouncedSetScrollTop = debounce(eventNode => {
+        setScrollTop(eventNode.scrollTop);
+      }, debounceTimer);
+
+      elem.addEventListener('scroll', () => debouncedSetScrollTop(elem));
+      elem.addEventListener('resize', () => debouncedSetScrollTop(elem));
+    }
+  }, []);
+
+  return (
+    <WindowingContext.Provider value={{ height, scrollTop }}>
```

Comment:
hmm is this more a getting an elements height & scrollTop rather than a windowing provider? (i get that you digest those for windowing, but this isnt doing any windowing logic).

I wonder if it could be more useful as a broader getElementBoundsProvider which returns the element bounds and positioning?

Reply:
Good point, with windowing I think it's better to keep it all bundled together for now.

We actually have a `RefValues` like component in our consuming app...maybe it needs to be brought over to this repo and seen if we can use it here.

Reply:
Agreed!